### PR TITLE
"make verify-config" target reports issues

### LINKED ISSUE #486: "make verify-config" target reports issues
**Describe the bug**
Run "make verify-config" and observe that examples/config-example.yaml is modified

{"level":"warn","error":"config: invalid config","time":"2022-03-29T00:01:16+03:00","message":"config dist-spec version: 0.1.0-dev differs from version actually used: 1.0.1, will be corrected automatically"}
...
{"level":"warn","error":"config: invalid config","time":"2022-03-29T00:01:17+03:00","message":"garbage-collect delay specified without enabling garbage-collect, will be ignored"}

**To Reproduce**
make verify-config

**Expected behavior**
All examples files should be reported as valid

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**

**Which issue does this PR fix**:
[Issue #486 ](https://github.com/project-zot/zot/issues/486)
Closes #486 
**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by laurentiuNiculae** [comment]
I have added make target that check for warnings when verifying configurations. This will force us to push only valid configurations.

**Comment by rchincha** [comment]
Why is `make verify-config` even changing files???
It is supposed to be a read-only operation.

**Comment by laurentiuNiculae** [comment]
When the distribution spec specified in the config file is wrong it will try to correct it by rewriting it. I made that change following [this](https://github.com/project-zot/zot/pull/448#discussion_r823058417) conversation.

File: pkg/cli/root.go [hunk 1]
```diff
@@ -354,22 +354,21 @@ func applyDefaultValues(config *config.Config, viperInstance *viper.Viper) {
 		}
 	}
 
-	if config.DistSpecVersion != distspec.Version {
-		log.Warn().Err(errors.ErrBadConfig).
-			Msgf("config dist-spec version: %s differs from version actually used: %s, will be corrected automatically",
-				config.DistSpecVersion, distspec.Version)
+	if !config.Storage.GC && viperInstance.Get("storage::gcdelay") == nil {
```
Comment (rchincha) [inline pkg/cli/root.go:357]: `storage::gcdelay`
^ interesting that I can scope like so.

File: pkg/cli/root.go [hunk 2]
```diff
@@ -354,22 +354,21 @@ func applyDefaultValues(config *config.Config, viperInstance *viper.Viper) {
...
+}
 
-		// rewrite the config file
-		viperInstance.Set("distSpecVersion", distspec.Version)
+func updateDistSpecVersion(config *config.Config) {
+	if config.DistSpecVersion == distspec.Version {
+		return
+	}
 
-		err := viperInstance.WriteConfig()
```
Comment (rchincha) [inline pkg/cli/root.go:365]: We should never have committed this. `verify` cannot modify config.

**Review by rchincha — DISMISSED** [review]
lgtm

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**